handler/transport/probe.go

```diff
@@ -58,58 +55,43 @@ func (state state) Print(f int, ft int) string {
 	return state.String()
 }
 
-func (b *Backend) getConfig() (*Config, error) {
-	req, _ := http.NewRequest(http.MethodGet, "", nil)
-	origin := b.transportConf.Scheme + "://" + b.transportConf.Origin
-	req = req.WithContext(context.WithValue(context.Background(), request.URLAttribute, origin))
-	return b.evalTransport(req)
-}
-
 func (b *Backend) NewProbe() {
 	p := &Probe{
-		name:             b.name,
-		failureThreshold: b.transportConf.HealthCheck.FailureThreshold,
-		time:             b.transportConf.HealthCheck.Period,
-		timeOut:          b.transportConf.HealthCheck.Timeout,
+		Name: b.name,
+		Opts: b.options.ParsedOptions,
+		Req:  b.options.Request,
 
-		counter: 0,
-		failure: 0,
+		State: StateInvalid,
 	}
-	probe.SetBackendProbe(p.name, StateInvalid.String())
-	c, err := b.getConfig()
-	if err != nil {
-		b.upstreamLog.LogEntry().WithError(err).Error()
-		return
-	}
-	p.config = c
 	go p.probe()
 }
 
 func (p *Probe) probe() {
-	req, _ := http.NewRequest(http.MethodGet, p.config.Scheme+"://"+p.config.Origin, nil)
-
 	for {
-		ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(p.timeOut))
-		res, err := http.DefaultClient.Do(req.WithContext(ctx))
+		ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(p.Opts.Timeout))
+		res, err := http.DefaultClient.Do(p.Req.WithContext(ctx))
```

Comment:
Maybe you could make use of the logger (given to the backend) to at least debug log those request errors.
Putting the log next to your current `//print...`:90 you could also provide the current state.